src/storage/src/hummock/store/version.rs

```diff
@@ -249,19 +249,24 @@ impl HummockReadVersion {
         Self {
             table_id,
             instance_id,
-            table_watermarks: committed_version.table_watermarks.get(&table_id).map(
-                |table_watermarks| {
-                    TableWatermarksIndex::new_committed(
-                        table_watermarks.clone(),
-                        committed_version
-                            .state_table_info
-                            .info()
-                            .get(&table_id)
-                            .expect("should exist")
-                            .committed_epoch,
-                    )
-                },
-            ),
+            table_watermarks: {
+                match committed_version.table_watermarks.get(&table_id) {
+                    Some(table_watermarks) => match table_watermarks.watermark_type {
+                        WatermarkSerdeType::PkPrefix => Some(TableWatermarksIndex::new_committed(
+                            table_watermarks.clone(),
+                            committed_version
+                                .state_table_info
+                                .info()
+                                .get(&table_id)
+                                .expect("should exist")
+                                .committed_epoch,
+                        )),
+
+                        WatermarkSerdeType::NonPkPrefix => None, /* do not fill the non-pk prefix watermark to index */
```

Comment:
Given that non pk prefix watermark will never be used in CN, should we exclude it in meta-CN version delta notification? We can add sanity check in the CN side to make sure this assumption holds.

Reply:
https://github.com/risingwavelabs/risingwave/blob/3500061180b4eafb23dbeed1c41c0ee7d00ae97d/src/storage/src/hummock/store/local_hummock_storage.rs#L532

Yes, we have avoided sending the NonPkPrefix Watermark in notifications

Reply:
I guess you pasted the wrong codes? Where do we exclude sending NonPkPrefix watermark in notifications? I cannot find the relevant codes. Also, if we do exlude that from notification, we should add sanity check for `WatermarkSerdeType::NonPkPrefix` the match arms in the CN side instead of filtering it out.